fc/parsing/actions.py

```diff
@@ -882,6 +876,178 @@ def _expr(self):
         self.rule = self.tokens[-1]
 
 
+class ProtocolVariable():
+    """
+    Collates information about a (new or existing) model variable used in a protocol.
+
+    Properties:
+
+    ``name``
+        A prefixed ontology term or a local variable name.
+    ``short_name``
+        A short name (no prefixes) that can be used for this variable.
+    ``long_name``
+        This variable's name plus any known aliases.
+    ``input_terms``
+        A list of rdf terms by which this variable is known as an input. This is typically zero or one term, but could
+        end up being more if two ontology terms specify the same model variables. For inputs this is an error.
+    ``output_terms``
+        A list of rdf terms by which this variable is known as a scalar output. This is typically zero or one term, but
+        multiple ontology terms may annotate the same output (as long as their units are consistent).
+    ``vector_output_terms``
+        A list of rdf terms by which this variable is known as a vector output. This is typically zero or one term.
+    ``is_input``
+        ``True`` iff the associated model variable will be set as a protocol input.
+    ``is_output``
+        ``True`` iff the associated model variable(s) will be read as a protocol output.
```

Comment:
Similarly 'read as a model output by the protocol'? Since the protocol itself also has inputs and outputs, which are different things!